### PR TITLE
Adding Microsoft Login

### PR DESCRIPTION
**Pull request by kfcampbell**
The entire site will now be wrapped with Microsoft login. This is a regression from our previous state where we had public/private routes, but not a permanent one. The library `react-aad-msal` supports public/private routes, so we can add that later whenever we'd like.

The user's name will be read from the Microsoft login claims and used as the display name for the PlayFab scoreboard. The IdToken and AccessToken are stored in local storage for us to use for server auth in the next feature!

### CODE REVIEW

File: src/utils/msalAuthProvider.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  cache: {
+    cacheLocation: "localStorage",
+    storeAuthStateInCookie: true
+  }
+};
+
+const authenticationParameters = {
+  scopes: [
+    "user.read"
+    //'https://<your-tenant-name>.onmicrosoft.com/<your-application-name>/<scope (i.e. demo.read)>'
```
Comment (Stephanie-Dover) [inline src/utils/msalAuthProvider.js]: can we remove these comments?

File: src/utils/msalAuthProvider.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { MsalAuthProvider, LoginType } from "react-aad-msal";
+const { REACT_APP_CLIENT_ID } = process.env;
+
+const config = {
+  auth: {
+    authority: "https://login.microsoftonline.com/common",
+    clientId: REACT_APP_CLIENT_ID
+    //redirectUri: '<OPTIONAL REDIRECT URI'
```
Comment (Stephanie-Dover) [inline src/utils/msalAuthProvider.js]: and this one?

File: src/utils/msalAuthProvider.js (hunk 1 above)
Comment (kfcampbell) [reply src/utils/msalAuthProvider.js]: Absolutely! I'll do it shortly.

**Comment by DanceParty** [comment]
Looks fantastic 🌮 💯 🏎 🤙

**Comment by DanceParty** [comment]
Unfortunately not relevant anymore

**Pull request closed without merging**